Tests/SwiftRedisTests/TestSetCommandsPart2.swift

```diff
@@ -0,0 +1,1150 @@
+            ("test_smove_redis", test_smove_redis),
+            ("test_spop", test_spop),
+            ("test_spop_redis", test_spop_redis),
+            ("test_srandmember", test_srandmember),
+            ("test_srandmember_redis", test_srandmember_redis),
+            ("test_srem", test_srem),
+            ("test_srem_redis", test_srem_redis),
+            ("test_sunion", test_sunion),
+            ("test_sunion_redis", test_sunion_redis),
+            ("test_sscan", test_sscan),
+            ("test_sscan_redis", test_sscan_redis),
+        ]
+    }
+    
+    let key1 = "key1"
+    let key2 = "key2"
+    let key3 = "key3"
+    let key4 = "key4"
+    let redisKey1 = RedisString("rediskey1")
+    let redisKey2 = RedisString("rediskey2")
+    let redisKey3 = RedisString("rediskey3")
+    let redisKey4 = RedisString("rediskey4")
+    
+    let member1 = "its over 9000"
+    let member2 = "What up doc?"
+    let member3 = "insert reference here"
+    let member4 = "insert test string here"
+    let redismember1 = RedisString("redis is over 9000")
+    let redismember2 = RedisString("What up doc? Just redis")
+    let redismember3 = RedisString("insert reference with redis here")
+    let redismember4 = RedisString("insert redisstring here")
+    
+    let redisVersions = ["2.6", "2.8", "3.0"]
+    
+    func setupTests(callback: () -> Void) {
+        connectRedis() {(error: NSError?) in
+            XCTAssertNil(error, "\(error != nil ? error!.localizedDescription : "")")
+            
+            redis.del(self.key1, self.key2, self.key3, self.key4) {(deleted: Int?, error: NSError?) in
+                callback()
+            }
+        }
+    }
+    
+    func setupRedisTests(callback: () -> Void) {
+        connectRedis() {(error: NSError?) in
+            XCTAssertNil(error, "\(error != nil ? error!.localizedDescription : "")")
+            
+            redis.del(self.redisKey1, self.redisKey2, self.redisKey3, self.redisKey4) {(deleted: Int?, error: NSError?) in
+                callback()
+            }
+        }
+    }
+    
+    func test_sadd() {
+        let expectation1 = expectation(description: "Add the specified members to the set stored at key")
+        setupTests {
+            
+            redis.sadd(self.key1, members: self.member1, self.member2, self.member3) {
+                (retrievedTotalElementAdded: Int?, error: NSError?) in
+                
+                XCTAssertNil(error)
+                XCTAssertNotNil(retrievedTotalElementAdded)
+                XCTAssertEqual(retrievedTotalElementAdded, 3)
+            }
+            
+            redis.sadd(self.key1, members: self.member1, self.member4) {
+                (retrievedTotalElementAdded: Int?, error: NSError?) in
+                
+                XCTAssertNil(error)
+                XCTAssertNotNil(retrievedTotalElementAdded)
+                XCTAssertEqual(retrievedTotalElementAdded, 1)
+                
+            }
+            expectation1.fulfill()
```

Comment:
The tests in general seekm ok, with major issue, that perhaps we can solve later on.

The tests assume that the Kitura-redis APIs run synchronously. While this is true now, we will eventually change that. When that happens all of these tests will break as the line of code expectation1.fulfill() might run before any of the callbacks would run.

I think the series of tests must be run as a nested sequence. The second one invoked from the callback of the first one, the third one invoked from the callback of the second one, etc. In the end of the callback of the last test in the test group, call fulfill on the expectation.